src/cdomains/arrayDomain.ml

```diff
@@ -941,7 +1003,762 @@ struct
+      MaySet.remove i (compute_set max_size)
+    else
+      MaySet.remove i may_nulls_set
+
+  let may_nulls_filter cond may_nulls_set max_size =
+    if MaySet.is_top may_nulls_set then
+      MaySet.filter cond (compute_set max_size)
+    else
+      MaySet.filter cond may_nulls_set
+
+  let may_nulls_min_elt may_nulls_set =
+    if MaySet.is_top may_nulls_set then
+      Z.zero
+    else
+      MaySet.min_elt may_nulls_set
+end
+
+module NullByte (Val: LatticeWithNull) (Idx: IntDomain.Z): Str with type value = Val.t and type idx = Idx.t =
+struct
+  module MustNulls = SetDomain.Reverse (SetDomain.ToppedSet (IntDomain.BigInt) (struct let topname = "All Null" end))
+  module MayNulls = SetDomain.ToppedSet (IntDomain.BigInt) (struct let topname = "All Null" end)
+  (* (Must Null Set, May Null Set, Array Size) *)
+  include Lattice.Prod3 (MustNulls) (MayNulls) (Idx)
+
+  include HelperFunctionsIndexMustMaySets
+
+  let name () = "arrays containing null bytes"
+  type idx = Idx.t
+  type value = Val.t
+
+  type ret = Null | NotNull | Top
+
+  (* helper: returns Idx.maximal except for Overflows that are mapped to None *)
+  let idx_maximal i = match Idx.maximal i with
+    | Some i -> (try Some (Z.of_int (Z.to_int i)) with Z.Overflow -> None)
+    | None -> None
+
+  let get ?(checkBounds=true) (ask: VDQ.t) (must_nulls_set, may_nulls_set, size) (e, i) =
+    let all_indexes_must_null i max =
+      let rec check_all_indexes i =
+        if Z.gt i max then
+          true
+        else if MustNulls.mem i must_nulls_set then
+          check_all_indexes (Z.succ i)
+        else
+          false in
+      if MustNulls.is_bot must_nulls_set then
+        true
+      else if Z.lt (Z.of_int (MustNulls.cardinal must_nulls_set)) (Z.sub max i) then
+        false
+      else
+        check_all_indexes i in
+    let min interval = match Idx.minimal interval with
+      | Some min_num when Z.geq min_num Z.zero -> min_num
+      | _ -> Z.zero in (* assume worst case minimal natural number *)
+
+    let min_i = min i in
+    let max_i = idx_maximal i in
+    let min_size = min size in
+
+    match max_i, idx_maximal size with
+    (* if there is no maximum value in index interval *)
+    | None, _ ->
+      (* ... return NotNull if no i >= min_i in may_nulls_set *)
+      if not (MayNulls.exists (Z.leq min_i) may_nulls_set) then
+        NotNull
+        (* ... else return Top *)
+      else
+        Top
+    (* if there is no maximum size *)
+    | Some max_i, None when Z.geq max_i Z.zero ->
+      (* ... and maximum value in index interval < minimal size, return Null if all numbers in index interval are in must_nulls_set *)
+      if Z.lt max_i min_size && all_indexes_must_null min_i max_i then
+        Null
+        (* ... return NotNull if no number in index interval is in may_nulls_set *)
+      else if not (MayNulls.exists (fun x -> Z.geq x min_i && Z.leq x max_i) may_nulls_set) then
+        NotNull
+      else
+        Top
```

Comment:
According to coverage reports, this code is never exercised. I think we should definitely add such cases. 
https://coveralls.io/builds/62854622/source?filename=src%2Fcdomains%2FarrayDomain.ml

(I guess by accessing at unknown indexes?)

Reply:
I've added an example where there is no maximum value in the index interval, but I don't know how to write a short example where there is no maximum size as initialization by an `int` variable or non-constant length leads to an error. This case rather is here for completeness.